src/dialogs/generic/dialog-box.ts

```diff
@@ -42,33 +48,64 @@ class DialogConfirmation extends LitElement {
         <h2>
           ${this._params.title
             ? this._params.title
-            : this.hass.localize("ui.dialogs.confirmation.title")}
+            : this._params.confirmation &&
+              this.hass.localize("ui.dialogs.generic.title")}
         </h2>
         <paper-dialog-scrollable>
-          <p>${this._params.text}</p>
+          ${this._params.prompt
+            ? html`
+                <paper-input
+                  autofocus
+                  .value=${this._text}
+                  @value-changed=${this._textChanged}
+                  label=${this._params.inputLabel
+                    ? this._params.inputLabel
+                    : ""}
+                  type=${this._params.inputType
+                    ? this._params.inputType
+                    : "text"}
+                ></paper-input>
+              `
+            : this._params.text
+            ? html`
+                <p>${this._params.text}</p>
```

Comment:
You can display text when it is a prompt?

Reply:
I'll move this above since a prompt might want an explanation above the input